docs/gke-autopilot.md

```diff
@@ -0,0 +1,268 @@
+
+```text
+gcloud services enable container.googleapis.com --project $PROJECT_ID
+```
+
+#### Create the Autopilot cluster
+
+By creating an Autopilot cluster (with `create-auto`) you only need to give the
+cluster a name. You can't specify any options that configure cluster nodes.
+
+```text
+gcloud container clusters create-auto microservices-demo
+```
+
+Output:
+
+```text
+WARNING: Starting with version 1.18, clusters will have shielded GKE nodes by default.
+WARNING: The Pod address range limits the maximum size of the cluster. Please refer to https://cloud.google.com/kubernetes-engine/docs/how-to/flexible-pod-cidr to learn how to optimize IP address allocation.
+WARNING: Starting with version 1.19, newly created clusters and node-pools will have COS_CONTAINERD as the default node image when no image type is specified.
+Creating cluster auto-microservices-demo in us-central1...done.
+Created [https://container.googleapis.com/v1beta1/projects/my-project/zones/us-central1/clusters/microservices-demo].
+To inspect the contents of your cluster, go to: https://console.cloud.google.com/kubernetes/workload_/gcloud/us-central1/microservices-demo?project=my-project
+kubeconfig entry generated for microservices-demo.
+NAME                       LOCATION     MASTER_VERSION  MASTER_IP     MACHINE_TYPE  NODE_VERSION    NUM_NODES  STATUS
+microservices-demo         us-central1  1.20.8-gke.900  35.223.7.213  e2-medium     1.20.8-gke.900  3          RUNNING
+```
+You can safely ignore the warnings. These simply indicate the cluster was
+created with default values. Other than network settings, these values can't be
+overridden anyway for an Autopilot cluster.
+
+You can see that the cluster is initially configured with three
+conservatively-sized machines for the nodes. These nodes are distributed across
+three zones in the `us-central1` region.
+
+## Deploying the demo
+
+Now that you have an Autopilot cluster, you can deploy the Online Boutique using
+Kubernetes:
+
+```text
+kubectl apply -f release/kubernetes-manifests.yaml
+```
+
+The output should look similar to deploying the cluster in standard mode, with a
+notable exception: you may see warnings similar to this:
+
+```text
+Warning: Autopilot increased resource requests for Deployment default/checkoutservice to meet requirements. See http://g.co/gke/autopilot-resources.
+deployment.apps/checkoutservice created
+service/checkoutservice created
+```
+
+Again, this is nothing to be concerned about, but it is notable that Autopilot
+will automatically adjust pod resource requirements to fit within [allowable
+resource ranges].
+
+Once the application has finished deploying, you can get the public IP with the
+following command:
+
+```text
+PUBLIC_IP=$(kubectl get service frontend-external -o jsonpath="{.status.loadBalancer.ingress[0].ip}")
+```
+
+## Wrapup: Why use GKE Autopilot mode?
+
+Most developers are familiar with **standard mode** since **Autopilot mode** is
```

Comment:
Let's capitalize "standard" (i.e., "Standard") since "Autopilot" is capitalized here.